packages/shared/src/hooks/useTimedAnimation.ts

```diff
@@ -0,0 +1,87 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { isNullOrUndefined } from '../lib/func';
+
+interface UseTimedAnimation {
+  timer: number;
+  isAnimating: boolean;
+  endAnimation: () => void;
+}
+
+interface UseTimedAnimationProps {
+  autoEndAnimation: boolean;
+  animationDuration: number;
+  onAnimationEnd?: () => void;
+  onAnimationStop?: () => void;
+}
+
+const TEMPORARY_ID = 1;
+const INTERVAL_COUNT = 10;
+const IN_OUT_ANIMATION = 140;
+
+export const useTimedAnimation = ({
```

Comment:
I see that we use a timer to indicate when the animation is done, and an interval to update the progress.
May I suggest using only an interval and when the animation is done trigger it from there?
This will make it less complicated and more synced I think

Reply:
If we will be removing the `timer` we won't have any reference of when to stop the interval itself as intervals run indefinitely until explicitly stopped.

Reply:
If we will be removing the timer we won't have any reference of when to stop the interval itself as intervals run indefinitely until explicitly stopped.

Reply:
But you use `setTimer` to keep track of the progress, don't you?

Reply:
Yes. Since we need the re-render so the component would know the change of value to display.

Reply:
Ah. And replace the interval to do it. Let me see how we can achieve it since we set value of 0 for unending animations. I'll check how it would turn out if we do -1.